troubadix/plugins/vt_file_permission.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2021 Greenbone Networks GmbH
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Iterator
+
+from troubadix.plugin import FilePlugin, LinterError, LinterResult
+
+
+class CheckVTFilePermissions(FilePlugin):
+    """This script checks wether the nasl file
+    has the correct file permissions
+    """
+
+    name = "check_vt_file_permissions"
+
+    def run(self) -> Iterator[LinterResult]:
+
+        permissions = oct(self.context.nasl_file.stat().st_mode)[-3:]
+
+        if permissions != "664":
```

Comment:
```suggestion
        if permissions != "644":
```
See ticket and git support :)

Reply:
The user should get a message which permissions are expected and why here too

Reply:
Nasl scripts are just expected to have `644` / are required not to be executable.

Original correction: Solved.